### PR TITLE
GeometryOptimizer input section

### PR DESCRIPTION
**Pull request by stigrj**
@moritzgubler 

Added a `GeometryOptimizer` input section containing the parameters you are using in the optimizer:

```
GeometryOptimizer {
    run = true
    max_iter = 100
    max_history_length = 10
    init_step_size = -0.5
    minimal_step_size = 0.01
    subspace_tolerance = 0.001
    max_force_component = 5.0e-3
}
```

By setting `run = false` in this section the standard single-point energy calculation is performed instead. You might have to nuke your old `build` and `install` directories and compile from scratch for the changes to take effect.

Let me know if you encounter any problems.

### CODE REVIEW

File: python/template.yml [hunk 1]
```diff
@@ -938,6 +938,46 @@ sections:
             docstring: |
               Formulation of the Permittivity function. Currently only the
               exponential is used.
+  - name: GeometryOptimizer
```
Comment (moritzgubler) [inline python/template.yml:941]: Is this the correct place for a detailed description of the sqnm parameters? If so, I can write a couple of sentences for each keyword.

**Comment by moritzgubler** [comment]
Thanks, @stigrj that looks great. I will test it quickly today or tomorrow.

**Comment by moritzgubler** [comment]
In my tests I was able to decrease the number of scf iterations by using the mw initial guess method after the first geometry optimization iteration. As an initial guess I used the orbitals from the previous geometry optimization iteration. This mechanism only works if ```path_orbitals = initial_guess``` and ```write_orbitals = true``` are set. The defaults are not set like that when a geometry optimization is done.

**Comment by moritzgubler** [comment]
To change the initial_guess method after the first iteration I directly adjust the cpp json [here](https://github.com/moritzgubler/mrchem/blob/3d6a9c0c0480da79e37ac5e2e69054e48d4ad64e/src/vc_sqnm/mrchem_optimizer.hpp#L126). Is it okay to do that like that? I feel like this may be a bit unsafe. If you think this is a bad idea, we can also remove this mechanism, the speed up is not that big.

**Comment by moritzgubler** [comment]
Hi @stigrj 
Did you already have time to look at my comments?
Best,
Moritz

**Comment by stigrj** [comment]
Sorry for the delay, I will have a closer look tonight

**Comment by stigrj** [comment]
I made the initial guess choice a bit more robust. I added a new input parameter
```
GeometryOptimizer {
  use_previous_guess = true
}
```
which toggles on the necessary keywords and changes the proper paths. By default it uses the `orbitals/` directory for these intermediate orbitals so that it does not interfere with the "actual" initial guess for the calculations, which would be located in the `initial_guess/` directory. If `use_previous_guess = false`, then it uses the same `SCF.guess_type` at every iteration.

File: python/template.yml (hunk 1 above)
Comment (stigrj) [reply python/template.yml:941]: Yes, please write a few words on each of them here

**Comment by moritzgubler** [comment]
perfect, I think we can merge this now.

File: python/template.yml (hunk 1 above)
Comment (moritzgubler) [reply python/template.yml:941]: Ok, I will do that on my repository after I accepted the pull request

**Pull request merged**